camel/functions/open_api_function.py

```diff
@@ -0,0 +1,383 @@
+
+    Args:
+        api_name (str): The name of the API, used to prefix generated function
+            names.
+        openapi_spec (Dict[str, Any]): The OpenAPI specification as a
+            dictionary.
+
+    Returns:
+        List[Dict[str, Any]]: A list of dictionaries, each representing a
+            function in the OpenAI schema format, including details about the
+            function's name, description, and parameters.
+
+    Raises:
+        ValueError: If an operation in the OpenAPI specification does not have
+            a description or summary.
+
+    Note:
+        This function assumes that the OpenAPI specification follows the 3.0+
+            format.
+
+    Reference:
+        https://swagger.io/specification/
+    """
+    result = []
+
+    for path, path_item in openapi_spec.get('paths', {}).items():
+        for method, op in path_item.items():
+            if op.get('deprecated') is True:
+                continue
+
+            # Get the function name from the operationId
+            # or construct it from the API method, and path
+            function_name = f"{api_name}"
+            operation_id = op.get('operationId')
+            if operation_id:
+                function_name += f"_{operation_id}"
+            else:
+                function_name += f"{method}{path.replace('/', '_')}"
+
+            description = op.get('description') or op.get('summary')
+            if not description:
+                error_message = (
+                    f"{method} {path} Operation from {api_name} "
+                    "does not have a description or summary."
+                )
+                raise ValueError(error_message)
+
+            description += f" This function is from {api_name} API."
+
+            # If the OpenAPI spec has a description,
+            # add it to the operation description
+            if 'description' in openapi_spec.get('info', {}):
+                description += f" {openapi_spec['info']['description']}"
+
+            # Get the parameters for the operation, if any
+            params = op.get('parameters', [])
+            properties: Dict[str, Any] = {}
+            required = []
+
+            for param in params:
+                if not param.get('deprecated', False):
+                    param_name = param['name'] + '_in_' + param['in']
+                    properties[param_name] = {}
+
+                    if 'description' in param:
+                        properties[param_name]['description'] = param[
+                            'description'
+                        ]
+
+                    if 'schema' in param:
+                        if (
+                            properties[param_name].get('description')
+                            is not None
+                            and 'description' in param['schema']
+                        ):
+                            param['schema'].pop('description')
+                        properties[param_name].update(param['schema'])
```

Comment:
```suggestion
                if 'description' in param:
                    properties[param_name]['description'] = param['description']
                elif 'description' in param.get('schema', {}):
                    properties[param_name]['description'] = param['schema']['description']
```
Please confirm whether the logic is same as suggested here

Reply:
OK, and for this part, I aims to deal with these situations:
1. if `param['description']` exists and not `param['schema']['description']`, use `param['description']` as description.
2. if not `param['description']` and `param['schema']['description']` exists, use `param['schema']['description']` as description.
3. if  `param['description']` and `param['schema']['description']` both exist, use `param['description']` as description.
4. if not `param['description']` and `param['schema']['description']` either, raise error.

So I think a revise edition is, 
```
if 'description' in param:
    properties[param_name]['description'] = param['description']

if 'schema' in param:
    if (properties[param_name].get('description') is not None and
            'description' in param['schema']):
        param['schema'].pop('description')
    properties[param_name].update(param['schema'])
```
WDYT?

Reply:
Yeah you are right, thanks for the explaination!

Reply:
Due to ruff `poetry run ruff format .` , after formatting, it will be changed to 
```
                    if 'description' in param:
                        properties[param_name]['description'] = param[
                            'description'
                        ]

                    if 'schema' in param:
                        if (
                            properties[param_name].get('description')
                            and 'description' in param['schema']
                        ):
                            param['schema'].pop('description')
                        properties[param_name].update(param['schema'])
```
again.